clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1581,6 +1581,23 @@ public void commitAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, Of
      * is invoked for the same partition more than once, the latest offset will be used on the next poll(). Note that
      * you may lose data if this API is arbitrarily used in the middle of consumption, to reset the fetch offsets
      *
+     * The next Consumer Record which will be retrieved when poll() is invoked will either have the offset specified or,
+     * by default, with {@link ConsumerConfig AUTO_RESET_CONFIG} set to "latest"
+     * a higher numbered offset. A higher numbered offset will be returned if there is no consumer record with the offset
+     * specified but there is one with a higher offset.
+     * seek(0) is equivalent to seek to beginning for a topic with beginning offset 0.
+     * seekToEnd() is equivalent to seeking to the highest known offset + 1.
+     *
+     * Seeking past the end of the highest known offset means an invalid offset is reached.
+     * Invalid offset behaviour is controlled by
+     * the {@link ConsumerConfig AUTO_RESET_CONFIG} property. If this is set to "earliest", the next poll() will seek to the beginning
+     * before returning a record. If it is set to "latest", it will seek to the last known record (similar to seekToEnd()).
+     * Note that, in the "latest" if any new records show up before {@link ConsumerConfig FETCH_MAX_WAIT_MS_CONFIG}, then
+     * the next poll() may succeed in seeking to the requested offset, or otherwise it will seek to the new latest known offset.
```

Comment:
I'm thinking we don't need to mention it here. First, it's not just `latest` case, `earlier` will also has this behavior. Second, it's too detail to users to know about it, and make users more confused about it, like so, how should we do about it? Third, so, back to the point you tried to mentioned, I think you're trying to explain the case:
1. current log end offset is 10
2. seek to 17
3. poll -> got nothing
4. produce 10 more records
5. poll -> got records offset starting from 17

Is that right? I think that's expected from user's point of view. User tried to seek to offset 17 and returning records starting from 17, everything works fine, isn't it? 

I think if you really want to talk about it, maybe you can say: the seek offset won't change to the in-flight fetch request, it will take effect in next fetch request. 

WDYT?

IMO, we don't need to mention it.

Reply:
i think this makes sense, but I'd like to leave in the note about fetch_max_wait_ms because that confused me when I was writing the tests